components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2839,34 +2856,89 @@ private ProvisioningConnectorConfig createProvisioningConnectorConfig(String out
      * @return Federated authenticator config of the specified ID.
      */
     private FederatedAuthenticatorConfig updateFederatedAuthenticatorConfig(String federatedAuthenticatorId,
```

Comment:
Given that based on the definedByType either system federated or user defined authenticator configs can be built better separate that to two.

Or can use a factory to get different authenticator configs built based on the type

Reply:
Addressed